toolchain/check/handle_function.cpp

```diff
@@ -364,15 +364,26 @@ static auto HandleFunctionDefinitionAfterSignature(
         SemIR::Function::GetParamFromParamRefId(context.sem_ir(), param_ref_id);
 
     // The parameter types need to be complete.
-    context.TryToCompleteType(param_info.inst.type_id, [&] {
-      CARBON_DIAGNOSTIC(
-          IncompleteTypeInFunctionParam, Error,
-          "parameter has incomplete type `{0}` in function definition",
-          SemIR::TypeId);
-      return context.emitter().Build(param_info.inst_id,
-                                     IncompleteTypeInFunctionParam,
-                                     param_info.inst.type_id);
-    });
+    context.TryToCompleteType(
+        param_info.inst.type_id,
+        [&] {
+          CARBON_DIAGNOSTIC(
+              IncompleteTypeInFunctionParam, Error,
+              "parameter has incomplete type `{0}` in function definition",
+              SemIR::TypeId);
+          return context.emitter().Build(param_info.inst_id,
+                                         IncompleteTypeInFunctionParam,
+                                         param_info.inst.type_id);
+        },
+        [&] {
+          CARBON_DIAGNOSTIC(
+              AbstractTypeInFunctionParam, Error,
+              "parameter has abstract type `{0}` in function definition",
+              SemIR::TypeId);
+          return context.emitter().Build(param_info.inst_id,
+                                         AbstractTypeInFunctionParam,
+                                         param_info.inst.type_id);
+        });
```

Comment:
I think this case should be permitted because a parameter binding is a value binding (by default, and we don't support the `var` modifier yet).